app/dashboard/utils.py

```diff
@@ -0,0 +1,203 @@
+'''
+    Copyright (C) 2017 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+import json
+import subprocess
+import time
+
+import ipfsapi
+from dashboard.helpers import UnsupportedSchemaException, normalizeURL, process_bounty_changes, process_bounty_details
+from dashboard.models import Bounty
+from eth_utils import to_checksum_address
+from web3 import HTTPProvider, Web3
+from web3.exceptions import BadFunctionCallOutput
+from web3.utils.datastructures import HexBytes
+
+
+class BountyNotFoundException(Exception):
+    pass
+
+
+class UnsupportedNetworkException(Exception):
+    pass
+
+
+class IPFSCantConnectException(Exception):
+    pass
+
+
+def startIPFS():
+    print('starting IPFS')
+    subp = subprocess.Popen(["ipfs", "daemon"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
This makes me nervous. 😂 Moving forward, we might want to consider this being offloaded to a `boto3` call to spin up an ECS task running IPFS (in the event we end up using ECS).